### PR TITLE
Only allow 'application/vnd.api+json' content-type

### PR DESCRIPTION
**Pull request by carlos-alberto**
Ensure Flump works with this content type, and only this content type.

NOTE: This will be a breaking change for existing apis using `Content-Type: application/json`.

### CODE REVIEW

**Comment by carlos-alberto** [comment]
This should resolve #18.

**Comment by obmarg** [comment]
What's the behaviour right now if you don't pass `application/vnd.api+json`?

**Comment by carlos-alberto** [comment]
Currently you MUST pass `application/json` in order to PATCH or POST. If you don't pass anything, or pass `application/vnd+api.json` it won't parse.

**Comment by obmarg** [comment]
I meant after this change.

**Comment by carlos-alberto** [comment]
Ah, if you don't pass `application/vnd+api.json` it won't parse and you will get back a 400. I'll edit to return a `415 Unsupported Content-Type`?

**Comment by obmarg** [comment]
It looks to me more like it'd except and you'd get a 500.  But yes, we definitely want to return a 415 if we're rejecting based on content-type.

However: I'm not sure we should be this strict.  json api is already a pain to use without adding even more requirements onto the client.  Could we not just accept (at least some) other things?

**Comment by obmarg** [comment]
Speaking of "how does this behave"  - some tests to verify it's behaviour would be a good idea.

**Comment by carlos-alberto** [comment]
Yeah fair enough, probably a 500. 

This is true, it depends if we want to move slightly away from the defined jsonapi.

Currently adding tests for the `415`.

**Comment by obmarg** [comment]
The jsonapi spec doesn't say anything about servers rejecting "application/json" etc. that I can see.

It says clients should use `application/vnd+api.json`, but nothing about servers rejecting clients that don't.

**Comment by carlos-alberto** [comment]
Good point. I'll allow `application/json` then? 

Which also prevents this being a breaking change.

**Comment by obmarg** [comment]
`application/json`, maybe even no content-type?

**Comment by carlos-alberto** [comment]
🔁

File: test/methods/test_patch.py [hunk 1]
```diff
@@ -89,3 +100,13 @@ def test_patch_fails_with_wrong_id(flask_client):
...
 
     assert response.status_code == 404
+
+
+def test_patch_fails_with_bad_content_type(flask_client):
+    create_response = create_user(flask_client)
+    response = patch_user(
+        flask_client, create_response.json['data']['id'],
+        etag=create_response.headers['Etag'], mimetype='bad_mimetype'
+    )
```
Comment (obmarg) [inline test/methods/test_patch.py]: Should probably have a test in here with the application/vnd mimetype?

File: test/methods/test_post.py [hunk 2]
```diff
@@ -58,3 +63,8 @@ def test_post_fails_if_wrong_resource_type_specified(flask_client):
     }
     response = create_user(flask_client, data=data)
     assert response.status_code == 409
+
+
+def test_post_fails_if_wrong_content_type_used(flask_client):
```
Comment (obmarg) [inline test/methods/test_post.py]: Same in here?

**Comment by carlos-alberto** [comment]
They both use that one `application/vnd.api+json` by default. So the main `test_post` and `test_patch` tests use that mimetype.

**Comment by obmarg** [comment]
🔀

**Pull request merged**